app/service/data_svc.py

```diff
@@ -389,15 +412,34 @@ def __init__(self, dao):
                                                           info['cleanup'].strip().encode(
                                                               'utf-8')).decode() if info.get(
                                                           'cleanup') else None,
-                                                      payload=info.get('payload'), parser=info.get('parser'))
+                                                      payload=info.get('payload'), parser=info.get('parser'),
+                                                      relationships=fact_relationships)
                 await self._delete_stale_abilities(ab)
 
-    async def _save_ability_extras(self, identifier, payload, parser):
+    async def _relationships_from_yaml(self, ability, relationship_types=['requires', 'consequence']):
+        relationships = []
+        for type in relationship_types:
```

Comment:
don't use 'type' as a variable name - it overshadows the built-in python type variable